bootstrap/handlers/httpserver.go

```diff
@@ -142,3 +150,30 @@ func (b *HttpServer) BootstrapHandler(
 
 	return true
 }
+
+// RequestLimitMiddleware is a middleware function that limits the request body size to Service.MaxRequestSize in kilobytes
+func RequestLimitMiddleware(sizeLimit int64, lc logger.LoggingClient) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			switch r.Method {
+			case http.MethodPost, http.MethodPut, http.MethodPatch:
+				if sizeLimit > 0 && r.ContentLength > sizeLimit*1000 {
```

Comment:
one KB is 1024, not 1000
```suggestion
				if sizeLimit > 0 && r.ContentLength > sizeLimit*1024 {
```

Reply:
fixed, also open issue https://github.com/edgexfoundry/edgex-go/issues/3980 to fix `MaxEventSize` value.